### PR TITLE
Comment to keep API preview headers around

### PR DESCRIPTION
**Pull request by andygrunwald**
## What type of PR is this?

documentation

## What this PR does / why we need it:

API preview headers (media types) are used to enable new features in GitHub API.
These media types will be added to the API request as headers and used to enable particular features on GitHub API that are still in preview.
After some time, specific media types will be promoted (to a "stable" state).
From then on, the preview headers are not required anymore to activate the additional feature on GitHub.com's API. However, this API header might still be needed for users to run a GitHub Enterprise Server on-premise.
It's not uncommon for GitHub Enterprise Server customers to run older versions which would probably rely on the preview headers for some time.
While the header promotion is going out for GitHub.com, it may be some time before it even arrives in GitHub Enterprise Server.
We keep those preview headers around to avoid breaking older GitHub Enterprise Server versions. Additionally, an non-functional (preview) header doesn't create any side effects on GitHub Cloud version.

See https://github.com/google/go-github/pull/2125 for full context.

## Special notes for your reviewer:

This Pull Request was triggered out of a discussion from https://github.com/google/go-github/pull/2125

### Explicit comment

Personally, I prefer and see value in extensive comments that tell the story and the full context next to the code (like in this PR).
I can understand when this interrupts the code reading flow for others.
I do think that the added advantage (by the additional context) is bigger.

Please let me know what you think about it and if any change is required.

## Additional documentation, usage docs, etc.:

- https://github.com/google/go-github/pull/2125

### CODE REVIEW

File: github/github.go [hunk 1]
```diff
@@ -48,6 +48,21 @@ const (
...
+	// After some time, specific media types will be promoted (to a "stable" state).
+	// From then on, the preview headers are not required anymore to activate the additional
+	// feature on GitHub.com's API. However, this API header might still be needed for users to run a GitHub Enterprise Server
+	// on-premise.
+	// It's not uncommon for GitHub Enterprise Server customers to run older versions which
+	// would probably rely on the preview headers for some time.
+	// While the header promotion is going out for GitHub.com, it may be some time before it
+	// even arrives in GitHub Enterprise Server.
+	// We keep those preview headers around to avoid breaking older GitHub Enterprise Server
+	// versions. Additionally, an non-functional (preview) header doesn't create any side effects
```
Comment (gmlewis) [inline github/github.go:62]:
```suggestion
	// versions. Additionally, non-functional (preview) headers don't create any side effects
```

File: github/github.go [hunk 2]
```diff
@@ -48,6 +48,21 @@ const (
 	mediaTypeIssueImportAPI    = "application/vnd.github.golden-comet-preview+json"
 
 	// Media Type values to access preview APIs
+	// These media types will be added to the API request as headers
+	// and used to enable particular features on GitHub API that are still in preview.
+	// After some time, specific media types will be promoted (to a "stable" state).
+	// From then on, the preview headers are not required anymore to activate the additional
+	// feature on GitHub.com's API. However, this API header might still be needed for users to run a GitHub Enterprise Server
+	// on-premise.
```
Comment (gmlewis) [inline github/github.go:56]:
```suggestion
	// to run a GitHub Enterprise Server on-premise.
```

File: github/github.go [hunk 3]
```diff
@@ -48,6 +48,21 @@ const (
 	mediaTypeIssueImportAPI    = "application/vnd.github.golden-comet-preview+json"
 
 	// Media Type values to access preview APIs
+	// These media types will be added to the API request as headers
+	// and used to enable particular features on GitHub API that are still in preview.
+	// After some time, specific media types will be promoted (to a "stable" state).
+	// From then on, the preview headers are not required anymore to activate the additional
+	// feature on GitHub.com's API. However, this API header might still be needed for users to run a GitHub Enterprise Server
```
Comment (gmlewis) [inline github/github.go:55]:
```suggestion
	// feature on GitHub.com's API. However, this API header might still be needed for users
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @andygrunwald !
Just a couple tiny nits, then LGTM and can be merged.

**Comment by andygrunwald** [comment]
Thanks for the quick review @gmlewis.
I added your changes.

**Review by gmlewis — APPROVED** [review]
Thank you, @andygrunwald !
LGTM.
Merging.

**Pull request merged**